### PR TITLE
Pickers for Race, Ethnicity, Gender on population dashboards

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

**Review by awisema — CHANGES_REQUESTED** [review]
Some of the supporting data is sent off to other report that don't honor the new filters.

File: app/controllers/concerns/client_detail_reports.rb [hunk 1]
```diff
@@ -53,6 +53,28 @@ module ClientDetailReports
...
+      scope.joins(:client).
+        merge(GrdaWarehouse::Hud::Client.where(race => 1))
+    end
+
+    private def filter_for_ethnicity(scope)
+      return scope unless @filter.ethnicity
+
+      scope.joins(:client).
+        merge(GrdaWarehouse::Hud::Client.where(Ethnicity: @filter.ethnicity))
+    end
```
Comment (eanders) [inline app/controllers/concerns/client_detail_reports.rb:76]: These need to be fixed!  Merge will replace previous selections

File: app/controllers/warehouse_reports/first_time_homeless_controller.rb [hunk 2]
```diff
@@ -58,6 +58,9 @@ def first_time_homeless_within_range(project_type)
       scope = filter_for_coc_codes(scope)
       scope = filter_for_organizations(scope)
       scope = filter_for_projects(scope)
+      scope = filter_for_gender(scope)
+      scope = filter_for_race(scope)
+      scope = filter_for_ethnicity(scope)
       scope = scope.joins(:project).merge(GrdaWarehouse::Hud::Project.viewable_by(current_user))
```
Comment (eanders) [inline app/controllers/warehouse_reports/first_time_homeless_controller.rb:64]: Does this erase the project filter?

Comment (eanders) [reply app/controllers/warehouse_reports/first_time_homeless_controller.rb:64]: Confirmed it does not.

**Pull request merged**